packages/docs/modules/ROOT/pages/network-js.adoc

```diff
@@ -0,0 +1,302 @@
+
+NOTE: If you use Infura in your own dapp, you will need to create an account with Infura and setup an Infura Project (with corresponding Infura Project ID).  See https://infura.io[Infura website] to create an account.
+
+== Add a component to the dapp
+For the second iteration of our dapp we will move the display of the current Ethereum network to a component and show how components are re-rendered when changes are made, such as the network.
+
+To install MetaMask as our injected web3 provider (if you don't have an injected web3 provider installed already), follow the instructions on the https://metamask.io[MetaMask website] to install the browser extension.
+
+In the code below, our `Web3Info` component expects a web3Context.  The `networkId`, `networkName` and `providerName` are obtained from the web3Context and displayed in the component.
+
+In the `client/src` directory create a `components` directory. In the `components` directory create a `Web3Info` directory for our `Web3Info` component.  Then create an `index.js` file in the `Web3Info` directory with the following code:
+[source,js]
+----
+import React from 'react';
+
+export default function Web3Info(props) {
+  const { web3Context } = props;
+  const { networkId, networkName, providerName } = web3Context;
+
+  return (
+    <div>
+      <h3> {props.title} </h3>
+      <div>Network: {networkId ? `${networkId} – ${networkName}` : 'No connection'}</div>
+      <div>Provider: {providerName}</div>
+    </div>
+  );
+}
+----
+
+In our App we use our `Web3Info` component and provide a web3Context.
+
+On the `client/src/App.js` file, replace `App.js` in our react project with the following code:
+
+[source,js]
+----
+import React from 'react';
+import './App.css';
+
+import { useWeb3 } from '@openzeppelin/network/react';
+
+import Web3Info from './components/Web3Info/index.js';
+
+// const infuraProjectId = process.env.REACT_APP_INFURA_PROJECT_ID;
+const infuraProjectId = '95202223388e49f48b423ea50a70e336';
+
+function App() {
+
+  const web3Context = useWeb3(`wss://mainnet.infura.io/ws/v3/${infuraProjectId}`);
+
+  return (
+    <div className="App">
+      <div>
+        <h1>OpenZeppelin Network.js</h1>
+        <Web3Info title="Web3 Info" web3Context={web3Context} />
+      </div>
+    </div>
+  );
+}
+
+export default App;
+----
+
+Start our dapp again by running `npm start` from within the `client` directory. 
+
+*OpenZeppelin Network.js* re-renders React components on changes such as account, network, and connection.
+
+Try changing networks in MetaMask and *OpenZeppelin Network.js* will cause the component to update with the network ID and network name.  
+
+== Add request access to injected web3 to the dapp
+For the third iteration of our dapp we will add a mechanism request to access injected web3 and once the user has connected will display the address of the users account.
```

Comment:
I think it's worth explaining to people that wallets have this measure in place to prevent the leaking of account information without user consent. It'll be very helpful to give the motivation behind it.

Reply:
Added explanation and moved link to EIP-1102.

Would appreciate your review on the wording.